resolver_conn_wrapper.go

```diff
@@ -0,0 +1,125 @@
+/*
+ *
+ * Copyright 2017 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package grpc
+
+import (
+	"strings"
+
+	"google.golang.org/grpc/grpclog"
+	"google.golang.org/grpc/resolver"
+)
+
+// ccResolverWrapper is a wrapper on top of cc for resolvers.
+// It implements resolver.ClientConnection interface.
+type ccResolverWrapper struct {
+	cc       *ClientConn
+	resolver resolver.Resolver
+	addrCh   chan []resolver.Address
+	scCh     chan string
+	done     chan struct{}
+}
+
+// newCCResolverWrapper parses cc.target for scheme and gets the resolver
+// builder for this scheme. It then builds the resolver and starts the
+// monitoring goroutine for it.
+//
+// This function could return nil, nil, in tests for old behaviors.
+// TODO(bar) never return nil, nil when DNS becomes the default resolver.
+func newCCResolverWrapper(cc *ClientConn) (*ccResolverWrapper, error) {
+	var scheme string
+	targetSplitted := strings.Split(cc.target, "://")
+	if len(targetSplitted) >= 2 {
+		scheme = targetSplitted[0]
+	}
+	grpclog.Infof("dialing to target with scheme: %q", scheme)
```

Comment:
This new log line means that all existing code has `grpc` emit logs when it dials, and along with [line 56][1]   all existing code that dials with a host name and no specified scheme gets a second log line about the lack of resolver for `""` scheme.

This is a problem because grpc is used in CLI clients and having grpc log these internal, unimportant details makes the UX of these CLIs confusing; random logs are showing up, suggesting that there's connection errors when there aren't.

It would be good if these two lines could be removed.

[1]: https://github.com/grpc/grpc-go/pull/1506/files#diff-23602dad3b1120382f350f80541865d7R56